addons/web/static/src/core/utils/autoresize.js

```diff
@@ -13,29 +13,28 @@ import { useEffect } from "@odoo/owl";
  * @param {Ref} ref
  */
 export function useAutoresize(ref, options = {}) {
-    let resize = null;
-    useEffect(
-        (el) => {
-            if (el) {
-                resize = (el instanceof HTMLInputElement ? resizeInput : resizeTextArea).bind(
-                    null,
-                    el,
-                    options
-                );
-                el.addEventListener("input", resize);
-                return () => {
-                    el.removeEventListener("input", resize);
-                    resize = null;
-                };
-            }
-        },
-        () => [ref.el]
-    );
-    useEffect(() => {
-        if (resize) {
-            resize(ref.el, options);
+    let resize, attach, detach;
+    let active = options.enable || true;
+    useEffect(el => {
+        attach = () => {
+            const resizeEl = (el instanceof HTMLInputElement ? resizeInput : resizeTextArea);
+            resize = el ? resizeEl.bind(null, el, options) : null;
+            el?.addEventListener("input", resize);
+        };
+        detach = () => {
+            el?.removeEventListener("input", resize);
+            resize = null;
+        };
+        if (active) {
+            attach();
         }
-    });
+        return detach;
+    }, () => [ref.el]);
+    useEffect(() => resize?.(), () => [ref.el?.value]);
+    return {
+        enable: () => (active || (attach?.(), resize?.()), active = true),
```

Comment:
i honestly dislike the `?` operator and using short boolean expressions (that hides the fact that this is not really an expression, but some code). Also, the `,` operator here is particularly annoying. Short code is not better code. I am not smart, I need simple code. What's wrong with:
```js
() => {
  if (!active) {
    if (attach) {
      attach();
   }
   if (resize) {
      resize();
   }
  active = true;
  }
}
```
same comment for disable